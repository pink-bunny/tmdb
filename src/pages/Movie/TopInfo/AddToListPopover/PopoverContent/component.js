import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'antd';
import CreateNewListModal from '../../../../../components/CreateNewListModal';
import AddToListButton from '../AddToListButton';

const PopoverContent = ({
  list,
  addToList,
  closePopover,
  createListModalSubmit
}) => (
  <>
    <div>
      <CreateNewListModal
        handleSubmitList={createListModalSubmit}
        triggerComponent={Button}
        triggerProps={{
          onClick: closePopover,
          type: 'link',
          children: 'Create new list ...'
        }}
      />
    </div>

    {list && list.map((item) => (
      <div key={item.id}>
        <AddToListButton
          addToList={addToList}
          id={item.id}
          name={item.name}
        />
      </div>
    ))}
  </>
);

PopoverContent.propTypes = {
  createListModalSubmit: PropTypes.func.isRequired,
  list: PropTypes.arrayOf(PropTypes.object),
  addToList: PropTypes.func.isRequired,
  closePopover: PropTypes.func.isRequired
};

PopoverContent.defaultProps = {
  list: null
};

export default PopoverContent;
