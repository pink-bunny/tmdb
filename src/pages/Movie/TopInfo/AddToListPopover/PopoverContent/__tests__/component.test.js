import React from 'react';
import { shallow } from 'enzyme';

import PopoverContent from '../component';

describe('PopoverContent component', () => {
  const requiredProps = {
    addToList: jest.fn(),
    closePopover: jest.fn(),
    createListModalSubmit: jest.fn()
  };
  const wrapper = shallow(<PopoverContent {...requiredProps} />); /* eslint-disable-line react/jsx-props-no-spreading, max-len */

  it('matches snapshot with required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with optional props', () => {
    wrapper.setProps({
      list: [
        { id: 1, name: 'Mock name 1' },
        { id: 2, name: 'Mock name 2' }
      ]
    });
    expect(wrapper).toMatchSnapshot();
  });
});
