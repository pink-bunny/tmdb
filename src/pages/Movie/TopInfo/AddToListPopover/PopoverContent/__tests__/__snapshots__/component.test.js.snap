// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopoverContent component matches snapshot with optional props 1`] = `
<Fragment>
  <div>
    <WrappedComponentShell
      handleSubmitList={[MockFunction]}
      triggerComponent={[Function]}
      triggerProps={
        Object {
          "children": "Create new list ...",
          "onClick": [MockFunction],
          "type": "link",
        }
      }
    />
  </div>
  <div
    key="1"
  >
    <AddToListButton
      addToList={[MockFunction]}
      id={1}
      name="Mock name 1"
    />
  </div>
  <div
    key="2"
  >
    <AddToListButton
      addToList={[MockFunction]}
      id={2}
      name="Mock name 2"
    />
  </div>
</Fragment>
`;

exports[`PopoverContent component matches snapshot with required props 1`] = `
<Fragment>
  <div>
    <WrappedComponentShell
      handleSubmitList={[MockFunction]}
      triggerComponent={[Function]}
      triggerProps={
        Object {
          "children": "Create new list ...",
          "onClick": [MockFunction],
          "type": "link",
        }
      }
    />
  </div>
</Fragment>
`;
